nixos/modules/services/desktops/pipewire/pipewire.nix

```
@@ -332,7 +332,7 @@ in {
       {
         # JACK intentionally not checked, as PW-on-JACK setups are a thing that some people may want
         assertion = (cfg.alsa.enable || cfg.pulse.enable) -> cfg.audio.enable;
-        message = "Using PipeWire's ALSA/PulseAudio compatibility layers requires running PipeWire as the sound server. Set `services.pipewire.audio.enable` to true.";
+        message = "Using PipeWire's ALSA/PulseAudio compatibility layers requires running PipeWire as the sound server. Set `services.pipewire.audio.enable` to true, or set both `services.pipewire.pulse.enable` and `services.pipewire.alsa.enable` to false if you wish to use pipewire only for compositor.";
```

```suggestion
        message = "Using PipeWire's ALSA/PulseAudio compatibility layers requires running PipeWire as the sound server. Set `services.pipewire.audio.enable` to false, or set both `services.pipewire.pulse.enable` and `services.pipewire.alsa.enable` to false if you wish to use pipewire only for compositor.";
```

Did you mean `false` there?

---

> Did you mean `false` there?

No, note how that `audio.enable` stayed `true` in the diff. The syntax of the assertion's logic is not clear on a first glance, but if I understand correctly, the message is being thrown according to the following truth table:

| `audio.enable`     | `alsa.enable`      | `pulse.enable`     | `throw`ing?        |
|--------------------|--------------------|--------------------|--------------------|
| :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: | :x:                |
| :heavy_check_mark: | :heavy_check_mark: | :x:                | :x:                |
| :heavy_check_mark: | :x:                | :heavy_check_mark: | :x:                |
| :heavy_check_mark: | :x:                | :x:                | :heavy_check_mark: (unless e.g `jack.enable==true`) |
| :x:                | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
| :x:                | :heavy_check_mark: | :x:                | :heavy_check_mark: |
| :x:                | :x:                | :heavy_check_mark: | :heavy_check_mark: |
| :x:                | :x:                | :x:                | :x:, and this is in-communicated possibility                 |

---

That's a very good suggestion. Thanks!